### PR TITLE
Add unit tests for parsing for generic sink

### PR DESCRIPTION
**Pull request by Ja-Gk-00**
Added unit tests to check edge cases for parsing through the generic API.
Improves code coverage to around 98%.

### CODE REVIEW

File: tests/unit_tests/test_parse/test_generic_edge_cases.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            IRI("http://ex/o"),
+            DEFAULT_GRAPH_IDENTIFIER,
+        )
+    )
+    s.add(Quad(IRI("http://ex/sA"), IRI("http://ex/pA"), IRI("http://ex/oA"), g1))
+    s.add(Quad(IRI("http://ex/sB"), IRI("http://ex/pB"), IRI("http://ex/oB"), g2))
+    return s
+
+
+def test_parse_grouped_from_flat_triples() -> None:
```
Comment (adanilenka) [inline tests/unit_tests/test_parse/test_generic_edge_cases.py:75]: have a feeling that test_parse_grouped_from_flat_triples and two after that belong not to unit tests. maybe in integration tests? bc they read/write files

File: tests/unit_tests/test_parse/test_generic_edge_cases.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        stream_types=StreamTypes(),
+        lookup_preset=LookupPreset(),
+        params=StreamParameters(),
+    )
+    with pytest.raises(NotImplementedError):
+        list(parse_jelly_flat(io.BytesIO(b"data"), frames=iter(()), options=opts))
+
+
+def test_parse_flat_get_options(monkeypatch: pytest.MonkeyPatch) -> None:
+    called = {"flag": False}
```
[the remark this answers is not in the document]
Comment (Ja-Gk-00) [reply tests/unit_tests/test_parse/test_generic_edge_cases.py]: Had to make it a dict for mutability to later call and change it inside `dummy_options` function. However, it was just for a check and after all it's not necessary. Removed.

File: pyjelly/integrations/generic/parse.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ class GenericGraphsAdapter(GenericQuadsBaseAdapter):
 
     """
 
-    _graph_id: str | None
+    _graph_id: str | IRI | None
```
Comment (adanilenka) [inline pyjelly/integrations/generic/parse.py:125]: maybe let's put Node | None here?
bc adding only IRI seems unfair, and then  later we will just remove str from Node union

Comment (Ja-Gk-00) [reply pyjelly/integrations/generic/parse.py]: Cambiado.
